scripts/gz10/gz10.py

```diff
@@ -0,0 +1,145 @@
+import itertools
+import h5py
+import numpy as np
+
+# TODO: Add BibTeX citation
+# Find for instance the citation on arxiv or on the dataset repo/website
+_CITATION = """\
+@InProceedings{huggingface:dataset,
+title = {A great new dataset},
+author={huggingface, Inc.
+},
+year={2020}
+}
+"""
+
+# TODO: Add description of the dataset here
+# You can copy an official description
+_DESCRIPTION = """\
+The GZ10 catalog from Leung et al. (2018) is a dataset of 17,736 galaxies with labels from the Galaxy Zoo 2 project.
+The catalog includes the following features for each galaxy: right ascension, declination, redshift, and a label from the Galaxy Zoo 2 project.
+https://astronn.readthedocs.io/en/latest/galaxy10.html
+"""
+
+# TODO: Add a link to an official homepage for the dataset here
+_HOMEPAGE = ""
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = ""
+
+_VERSION = "0.0.1"
+
+
+# TODO: Name of the dataset usually matches the script name with CamelCase instead of snake_case
+class GZ10(datasets.GeneratorBasedBuilder):
+    """TODO: Short description of my dataset."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="gz10_with_healpix",
+                               version=VERSION,
+                               data_files=DataFilesPatternsDict.from_patterns({'train': ['datafiles/healpix=*/*.h5']}),
+                               description="GZ-10 Catalog loaded with Healpix indices (using NSIDE=16). Images not included."),
+        datasets.BuilderConfig(name="gz10_images",
+                                version=VERSION,
+                                data_files=DataFilesPatternsDict.from_patterns({'train': ['Galaxy10_DECals.h5']}),
+                                description="GZ-10 Catalog loaded from one HDF5 file."),
+        datasets.BuilderConfig(name="gz10_with_healpix_with_images",
+                               version=VERSION,
+                               data_files=DataFilesPatternsDict.from_patterns({'train': ['datafiles/healpix=*/*.h5']}),
+                               description="GZ-10 Catalog loaded with Healpix indices (using NSIDE=16). Images included.")
+    ]
+
+    DEFAULT_CONFIG_NAME = "gz10_with_healpix"  # It's not mandatory to have a default configuration. Just use one if it make sense.
+
+    _image_size = 256
+    _n_samples = 17736
+
+    def _info(self):
+        """Defines the dataset info."""
+        features = datasets.Features(
+            {
+                "gz10_label": datasets.Value("int32"),
+                "ra": datasets.Value("float32"),
+                "dec": datasets.Value("float32"),
+                "object_id": datasets.Value("string"),
+            }
+        )
+
+        if self.config.name == "gz10_images" or self.config.name == "gz10_with_healpix_with_images":
+            features["images"] = datasets.Array3D(shape=(self._image_size, self._image_size, 3), dtype="uint8")
```

Comment:
Is this different to the other astropile datasets? I get a little lost but I thought people were keen on recording the flux rather than the processed uint8 images.